pkgs/development/python-modules/ldap/default.nix

```
@@ -13,20 +14,22 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ pyasn1 pyasn1-modules ];
 
+  checkInputs = [ pytestCheckHook ];
   buildInputs = [ openldap cyrus_sasl ];
 
-  checkInputs = [ pytest ];
-
-  checkPhase = ''
+  preCheck = ''
     # Needed by tests to setup a mockup ldap server.
     export BIN="${openldap}/bin"
     export SBIN="${openldap}/bin"
     export SLAPD="${openldap}/libexec/slapd"
     export SCHEMA="${openldap}/etc/schema"
-
-    py.test
   '';
 
+  # See https://github.com/python-ldap/python-ldap/issues/407
+  disabledTests = lib.optionals (isPy39) [
```

```suggestion
  disabledTests = lib.optionals (pythonAtLeast "3.9") [
```
It will probably also break with 3.10

---

Well, hopefully they'll have a new release that fixes it :P

---

True, but you should at least get rid of the parentheses.